python/lsst/ts/MTAirCompressor/aircompressor_csc.py

```diff
@@ -178,51 +169,55 @@ def run_sim():
                 None, run_sim
             )
             await asyncio.sleep(2)
-            self.client = ModbusClient(host="localhost", port=5020)
-        else:
-            self.client = ModbusClient(host=self.hostname)
-
-        if self.client.connect() is False:
-            self.log.error(f"Cannot establish connection to {self.hostname}")
 
-        self.telemetry_task = asyncio.create_task(self.telemetry_loop())
+        try:
+            await self.connect()
+            if self.poll_task.done():
+                self.poll_task = asyncio.create_task(self.poll_loop())
+        except ModbusError as er:
+            await self.log_modbus_error(er, "Starting up:", True)
+            return
 
     async def end_enable(self, data):
         """Power on compressor after switching to enable state. Raise exception
-        if compressor cannot be powered on."""
-        poweredOn = self.client.write_register(
-            Registers.REMOTE_CMD, 0xFF01, unit=self.unit
-        )
-        if poweredOn.isError():
-            raise ModbusError(
-                "Cannot power on compressor", poweredOn, Registers.REMOTE_CMD
+        if compressor cannot be powered on. Ignore state transition triggered
+        by auto update."""
+        if self._status_update is True:
+            return
```

Comment:
flake8 recommends `if self._status_update:` and I agree (assuming this is a boolean flag).

Why is this a no-op if _status_update is true? What does that mean? It seems counter-intuitive. I would have expected this that flag to mean "I want a status update; so ask for one now". Perhaps it means "I have seen a status update since connecting"? Adding a comment to the constructor explaining what this flag means would be a help.

Reply:
This is to not power up compressor as it transition to enabled state. DM-35280 will solve that.